beacon/goclient/attest.go

```diff
@@ -27,22 +27,106 @@ func (gc *GoClient) AttesterDuties(ctx context.Context, epoch phase0.Epoch, vali
 	return resp.Data, nil
 }
 
-func (gc *GoClient) GetAttestationData(slot phase0.Slot, committeeIndex phase0.CommitteeIndex) (*phase0.AttestationData, spec.DataVersion, error) {
+// GetAttestationData returns attestation data for a given slot (which is same for all 64 committeeIndex
+// values that identify Ethereum committees chosen to attest on this slot).
+// Note, committeeIndex is an optional parameter that will be used to set AttestationData.Index
+// in the resulting data returned from this function.
+// Note, result returned is meant to be read-only, it's not safe to modify it (because it will be
+// accessed by multiple concurrent readers).
+func (gc *GoClient) GetAttestationData(slot phase0.Slot, committeeIndex phase0.CommitteeIndex) (
```

Comment:
Regarding `committeeIndex`,

as far as I can tell (looking at the usage in code), we don't really need `committeeIndex` to even exist as parameter for this func, removing it completely though might be not that simple because `spec` repo is also some definitions with it.

Reply:
@iurii-ssv 

> as far as I can tell (looking at the usage in code), we don't really need **committeeIndex** to even exist as parameter for this func, removing it completely though might be not that simple because `spec` repo is also some definitions with it.

`(*AttesterRunner).executeDuty` and `(*GoClient).SubmitAggregateSelectionProof` (from `(*AggregatorRunner).ProcessPreConsensus`) seem to call `GetAttestationData` with a real committee index, which is taken from duty, so we cannot remove it right now. But yeah, we don't always need it, so probably the optional pattern would suit better here, however, we'd need to change spec for this, which is not simple, so I think passing 0 where we don't need it and leaving a comment is fine.

Reply:
> (*AttesterRunner).executeDuty and (*GoClient).SubmitAggregateSelectionProof (from (*AggregatorRunner).ProcessPreConsensus) seem to call GetAttestationData with a real committee index, which is taken from duty, so we cannot remove it right now.

Yeah, I think it doesn't `committeeIndex` part or the result returned, only `attData.HashTreeRoot()`, 

but regardless, removing `committeeIndex` param is probably not worth it.

Reply:
@iurii-ssv It's used in `HashTreeRoot`, so changing it would change the hashing algorithm and make the branch incompatible with other branches. There may be operators running different versions

But yeah, if not this, we could get rid of it. Maybe we'll do it sometime in the future